### DISCUSSION TITLE
AvailableBackends.  numpy vs PYTORCH during compute_model

### DISCUSSION THREAD

**Discussion by DC-geo** [gempy-project/gempy#907] [Q&A] [answered]
Hi and thanks so much for making Gempy 3! (I can finally wrangle the installation with ease).

I'm getting through the tutorials and noticed in ch1_1 ln 217 when running:

```
gp.compute_model(geo_model)
```
i get:
```
Setting Backend To: AvailableBackends.numpy
Number of voxels marked as high curvature: 2 of 8
Number of voxels marked as high curvature: 11 of 64
Number of voxels marked as high curvature: 17 of 432
Solutions(4 Octree Levels, 5 DualContouringMeshes)
```

but the tutorial suggests it should be:

> Setting Backend To: AvailableBackends.PYTORCH

The outputs appear to be the same (image etc). So is this an issue?

**Comment by javoha** [comment, marked as answer]
Hi,
thanks for the question: To start with the short answer - No, this is not an issue. Gempy v3 offers to perform the model computation with different backends. Numpy is the most straightforward, easy to install and easy to understand version, making it the default. You can also install pytorch and use this backend (depending on you setup computing either on cpu or gpu) to run models faster. The result will be the same.
I think all the examples in the documentation ran with the torch backend to speed things up, but I would advice tp start with the numpy version at the beginning for smaller models.
Cheers
Jan

**Comment by DC-geo** [comment]
Thanks Jan! 
I'll have to work out how to set pytorch as the backend at some later date then.
